### ISSUE TITLE
Same block getting imported multiple times due to race condition

### ISSUE THREAD

**Issue by jimmygchen** [sigp/lighthouse#6439]
## Description

**TL;DR:** This is actually an existing race condition and can be triggered today. This can happen when blobs arrived from RPC and gossip around the same time and both triggering a block import. There's no impact other than extra work performed:  in the 2nd import, the state cache is expected to miss, as the previous import `pop`'d the state, so this will result in state reconstruction, which takes ~80ms.

### PeerDAS occurence

On `peer-das-devnet-2`, I'm seeing the same block getting written to the database multiple times (4 times within 2 ms!). This occurred on a supernode.

I think this is because we keep the `PendingComponent`s in the cache until we complete block import, so for each gossip component that made it through to the DA checker after the block is available, it would trigger another import:

https://github.com/sigp/lighthouse/blob/07afa6c2a57e5eff515482a62e064ff975e36966/beacon_node/beacon_chain/src/beacon_chain.rs#L3635-L3646

With PeerDAS supernodes, once we receive 64 columns, the remaining columns can come from reconstruction and block can be made available after this. However, we can still get columns from gossip (as they haven't been seen via gossip), and each of these gossip columns could trigger a block import, as the `PendingComponent` in the cache is "complete".

The change to keep `PendingComponent` in the DA cache was intentional and was made in the following PR to address an issue with sync lookup:
https://github.com/sigp/lighthouse/pull/5845

I'm not sure if there's a better way, but if we do need to keep the block in the DA Checker during import, perhaps we can do a check before trying to process a gossip block/blob/data_column  - if block has already been made available, just `return` instead of processing and re-importing.

**Comment by jimmygchen** [comment]
Logs from `Sep 25` on lighthouse-geth-3 on `peerdas-devnet-2`:
- `05:29:05.122`: received block `0x647586a23cd593120c0277581cbf829294821abdd6f8306564b78351be4961d4`
- `05:29:05.929`: received 67 data columns
- `05:29:06.476`: `writing data_columns to store` (block import), data column reconstruction must have occurred for this to happen (on the 64th data column recieved)
- `05:29:06.482`: another `writing data_columns to store` - this should be from the 65th gossip data columns
- `05:29:06.489`: `Gossipsub data column processed, imported fully available block` - this proved the above write is triggered from gossip column
- `05:29:06.489`: `Sending pubsub messages..` this indicates reconstruction is complete and publish a bunch of reconstructed data columns
- `05:29:06.489`: `Writing data_columns to store`: 3rd write, triggered from the 66th column recieved
- `05:29:06.506`: `Writing data_columns to store`: 4th write, triggered from the 67th column recieved
- `05:29:06.517`: Seeing the remaining columns recieved from gossip but ignored due to `DuplicateFullyImported`

**Comment by jimmygchen** [comment]
Looks like this is still happening in devnet-7, just noticed the block and columns got persisted 7 times. We should fix this.

**Comment by dapplion** [comment]
> if block has already been made available, just return instead of processing and re-importing.

Seems like a simple fix.

We can change `PendingComponents` into holding a State enum

```rust
enum State {
    WaitingComponents {
        block: Option<DietAvailabilityPendingExecutedBlock<E>>,
        blobs: RuntimeFixedVector<Option<KzgVerifiedBlob<E>>>,
        data_columns: Vec<KzgVerifiedCustodyDataColumn<E>>,
    }
    ReconstructionStarted,
    Importing,
}
```

**Comment by jimmygchen** [comment]
Related issue #7461 - solving that one will likely greatly reduce the likelihood of this one.

**Comment by jimmygchen** [comment]
While #7961 solves the bug with data inconsistency and likelihood of race due to lock contention, it doesn't eliminate block getting imported multiple times - the solution _accepted_ this trade off for simplicity. 

However after a chat with @ethDreamer today, I'm convinced that we _could_ avoid multiple import quite easily without much added complexity.

We didn't merge #7931 because of the concern commented [here](https://github.com/sigp/lighthouse/pull/7931#discussion_r2299767457):

> I'm worried about the case where block is available but is awaiting import / recompute head. This could be a while and we may not want to send BlockComponentProcessed event until then

^ if the block is already `available` when we attempt to insert a component, we simply don't do anything and can drop the responsibility of importing block, recomputing head and notifying sync - because another thread would be doing this anyways - so re-import or awaiting for the import not strictly unnecessary.
